src/NuGet.Deterministic.Package/build/NuGet.Deterministic.targets

```diff
@@ -1,65 +1,96 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <UsingTask AssemblyFile="$(NuGetDeterministicTasksAssemblyPath)" TaskName="NuGet.Tasks.GenerateLockedPackageConfigurationFile" />
 
-  <PropertyGroup>
-    <NuGetDeterministicExcludeExplicitPackages Condition="'$(NuGetDeterministicExcludeExplicitPackages)'==''">true</NuGetDeterministicExcludeExplicitPackages>
-    <GenerateNuGetDeterministicProps Condition=" '$(GenerateNuGetDeterministicProps)' == '' And '$(NuGetDeterministicPropsWasImported)' != 'true' And '$(ImportNuGetDeterministicProps)' != 'true' ">true</GenerateNuGetDeterministicProps>
-    <PrepareForBuildDependsOn Condition=" '$(GenerateNuGetDeterministicProps)' == 'true' ">$(PrepareForBuildDependsOn);GenerateNuGetDeterministicProps</PrepareForBuildDependsOn>
-    <PrepareForBuildDependsOn Condition=" '$(NuGetDeterministicPropsWasImported)' == 'true' Or '$(ImportNuGetDeterministicProps)' == 'true' ">$(PrepareForBuildDependsOn);ValidateNuGetDeterministic</PrepareForBuildDependsOn>
+  <PropertyGroup Condition=" '$(EnableNuGetDeterministic)' != 'false' ">
+    <NuGetDeterministicTasksAssemblyPath Condition=" '$(NuGetDeterministicTasksAssemblyPath)' == '' ">$(MSBuildThisFileDirectory)tasks\NuGet.Tasks.Deterministic.dll</NuGetDeterministicTasksAssemblyPath>
+    
+    <!--
+      The path to a file written to when validating to not run validation unless it is necessary.
+    -->
+    <NuGetDeterministicPackagesValidatedFile Condition=" '$(NuGetDeterministicPackagesValidatedFile)' == '' ">$(IntermediateOutputPath)$(MSBuildProjectFile).ndpv</NuGetDeterministicPackagesValidatedFile>
```

Comment:
in non cloudbuild scenario can we rely on IntermediateOutputPath being set yet?

Reply:
Good point.  I ended up removing this logic anyway since the validation only really happens once in a hosted build.  Not to mention that you were right that it was only happening once even on a multi-targeted build...